lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1084,31 +1112,12 @@ static bool performCompileStepsPostSILGen(CompilerInstance &Instance,
   // done, depending on the compiler setting.
 
   auto SerializeSILModuleAction = [&]() {
```

Comment:
Notice how this is already extracted; *this* is the thing you'd want to pull out, not the option computation.

Reply:
This is stored in an instance variable of the SILModule and has state (MSF, moduleIsPublic, etc.) associated with it. If the state is replicated in the SILModule, we could just get away with storing a function "pointer" there. Otherwise, we would have to instantiate a new class with the desired state and behavior, which is exactly what the lambda does. I guess I'll need more guidance here.

Reply:
Oops, right. I forgot what this thing was. I'll take another look soon.